projects/plugins/super-cache/wp-cache.php

```diff
@@ -3207,113 +3207,131 @@ function wp_cron_preload_cache() {
+					set_time_limit( 60 );
+					if ( $url === '' ) {
+						continue;
 					}
-					$fp = fopen( $taxonomy_filename, 'w' );
+
+					$url_info = wp_parse_url( $url );
+					$dir      = get_supercache_dir() . $url_info['path'];
+					wp_cache_debug( "wp_cron_preload_cache: delete $dir", 5 );
+					wpsc_delete_files( $dir );
+					prune_super_cache( trailingslashit( $dir ) . 'feed/', true );
+					prune_super_cache( trailingslashit( $dir ) . 'page/', true );
+					// phpcs:ignore -- WordPress.PHP.NoSilencedErrors.Discouraged WordPress.WP.AlternativeFunctions.file_system_operations_fopen
+					$fp = @fopen( $permalink_counter_msg, 'w' );
 					if ( $fp ) {
-						fwrite( $fp, implode( "\n", $details ) );
-						fclose( $fp );
+						// phpcs:ignore -- WordPress.PHP.NoSilencedErrors.Discouraged WordPress.WP.AlternativeFunctions.file_system_operations_fwrite
+						@fwrite( $fp, "$taxonomy: $url" );
+						// phpcs:ignore -- WordPress.WP.AlternativeFunctions.file_system_operations_fclose WordPress.PHP.NoSilencedErrors.Discouraged
+						@fclose( $fp );
+					}
+					wp_remote_get(
+						$url,
+						array(
+							'timeout'  => 60,
+							'blocking' => true,
+						)
+					);
+					wp_cache_debug( "wp_cron_preload_cache: fetched $url", 5 );
+					sleep( 1 );
+
+					// phpcs:ignore WordPress.PHP.NoSilencedErrors.Discouraged
+					if ( @file_exists( $cache_path . 'stop_preload.txt' ) ) {
+						wp_cache_debug( 'wp_cron_preload_cache: cancelling preload. stop_preload.txt found.', 5 );
+						wpsc_reset_preload_settings();
+
+						if ( $wp_cache_preload_email_me ) {
+							// translators: Home URL of website
+							wp_mail( get_option( 'admin_email' ), sprintf( __( '[%1$s] Cache Preload Stopped', 'wp-super-cache' ), home_url(), '' ), ' ' );
+						}
+						return true;
 					}
 				}
-
-				if (
-					$preload_more_taxonomies === false &&
-					count( $details ) > 0 &&
-					$details[0] !== ''
-				) {
-					$preload_more_taxonomies = true;
+				// phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_operations_fopen
+				$fp = fopen( $taxonomy_filename, 'w' );
+				if ( $fp ) {
+					// phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_operations_fwrite
+					fwrite( $fp, implode( "\n", $details ) );
+					// phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_operations_fclose
+					fclose( $fp );
 				}
 			}
 
-			if ( $preload_more_taxonomies === true ) {
-				wpsc_schedule_next_preload();
-				return true;
+			if (
+				$preload_more_taxonomies === false &&
+				count( $details ) > 0 &&
+				$details[0] !== ''
+			) {
+				$preload_more_taxonomies = true;
 			}
 		}
+
+		if ( $preload_more_taxonomies === true ) {
+			wpsc_schedule_next_preload();
+			return true;
+		}
+	} elseif ( $c === 0 && $wp_cache_preload_email_me ) {
+		// translators: Home URL of website, blank space
+		wp_mail( get_option( 'admin_email' ), sprintf( __( '[%1$s] Cache Preload Started', 'wp-super-cache' ), home_url(), '' ), ' ' );
```

Comment:
I noticed the email notification code has been added twice. Wouldn't it make more sense to send the notification when the mutex file is created? As I understand, that file is what prevents multiple preloads from happening at the same time.

Reply:
The mutex file is deleted at the end of a loop, and then recreated on the next scheduled loop, so it's impossible to know at the top of the function if this is the first or 10th loop, especially as I got rid of the taxonomy_preload transient option.

Reply:
Thanks, fixed that!